src/Build.UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -2790,6 +2790,37 @@ public void IsOsPlatformShouldBeCaseInsensitiveToParameter()
             Assert.Equal("True", result);
         }
 
+        [Theory]
+        [InlineData("vv1.0")]
+        [InlineData("NotAVersion")]
+        [InlineData("1.2.3.4~5")]
```

Comment:
Add a test for a bare number like `3`?

Reply:
```
Microsoft (R) Visual C# Interactive Compiler version 3.5.0-beta1-19558-01 ()
Copyright (C) Microsoft Corporation. All rights reserved.

Type "#help" for more information.
> System.Version.Parse("3")
System.ArgumentException: Version string portion was too short or too long.
```

Good catch. I think this should work, but basically requires that we do not use Version.Parse, but rather have our own parser. Which was something I was only wanting to do if there was a significant perf win.